content/post/094-js-ipfs-0.44.md

```diff
@@ -0,0 +1,177 @@
+
+## 🤖 Cancellable requests
+
+A user should be able to cancel long-lived asynchronous API operations. For example, if you are fetching the content for a [CID][] from the network, and that CID is not resolveable, you should be able to set a timeout value for the request, after which the content would no longer be sought and control would be returned to your code with an error message describing what happened.
+
+This is not as straightforward as it seems in JavaScript because a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) is returned from the API call to resolve a piece of content and the [Promises spec](https://promisesaplus.com/) includes nothing about cancelling a Promise.
+
+However, the browser [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) has a notion of an [AbortSignal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) which can be used to abort web requests. They are interacted with via an [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController):
+
+```javascript
+const controller = new AbortController()
+
+setTimeout(() => {
+  controller.abort()
+}, 1000)
+
+try {
+  await result = fetch('http://example.com', {
+    signal: controller.signal
+  })
+} catch (err) {
+  console.err(err) // prints: user aborted the request
+}
+```
+
+We’ve taken this approach and integrated it into every API call, so you can now use AbortControllers to cancel requests! We’ve also used this mechanism to add a `timeout` option to every API call which internally uses an AbortSignal to stop the request after the specified amount of time:
+
+```javascript
+const cid = new CID('QmWillNeverResolve')
+
+try {
+  await ipfs.get(cid, {
+    timeout: 1000 // abort after 1000ms
+  })
+} catch (err) {
+  console.err(err)
+}
+```
+
+You could also roll this yourself:
+
+```javascript
+const cid = new CID('QmWillNeverResolve')
+const controller = new AbortController()
+
+setTimeout(() => {
+  controller.abort()
+}, 1000)
+
+try {
+  await ipfs.get(cid, {
+    signal: controller.signal
+  })
+} catch (err) {
+  console.err(err)
+}
+```
+
+The AbortSignal is passed all the way down the call stack to the libp2p and ipld components that underpin IPFS. In this initial release, timeouts are supported by the IPFS layer and in future releases libp2p and ipld will use the AbortSignal to also free up resources and perform any other necessary cleanup that would have been done if the request had completed successfully.
+
+## New browser datastore
+
+`js-IPFS@0.44.0` brings a new datastore to the browser. In the browser, all blocks and other repo data is currently stored in [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API), given that it is the only way available to persist and query large amounts of user data at the time of writing.
```

Comment:
Do we not?

Reply:
Until we request the permission and call .persist(), etc it can be blown away at any point, even if some other origin hits the overall browser-level quota: ipfs/js-ipfs#2937

Reply:
I think the difference is largely semantic.  Anyway I didn't want to keep repeating the word 'store' in the sentence.